powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/v3/PowerAuthServiceImpl.java

```diff
@@ -142,6 +146,61 @@ public GetActivationListForUserResponse getActivationListForUser(GetActivationLi
         }
     }
 
+    @Override
+    @Transactional
+    public LookupActivationsResponse lookupActivations(LookupActivationsRequest request) throws GenericServiceException {
+        if (request.getUserIds().isEmpty()) {
+            logger.warn("Invalid request");
+            throw localizationProvider.buildExceptionForCode(ServiceError.INVALID_REQUEST);
+        }
+        try {
+            List<String> userIds = request.getUserIds();
+            List<Long> applicationIds = request.getApplicationIds();
+            Date timestampLastUsedBefore = null;
+            if (request.getTimestampLastUsedBefore() != null) {
+                timestampLastUsedBefore = XMLGregorianCalendarConverter.convertTo(request.getTimestampLastUsedBefore());
+            }
+            Date timestampLastUsedAfter = null;
+            if (request.getTimestampLastUsedAfter() != null) {
+                timestampLastUsedAfter = XMLGregorianCalendarConverter.convertTo(request.getTimestampLastUsedAfter());
+            }
+            ActivationStatus activationStatus = null;
+            if (request.getActivationStatus() != null) {
+                activationStatus = activationStatusConverter.convert(request.getActivationStatus());
+            }
+            logger.info("LookupActivationsRequest received");
+            LookupActivationsResponse response = behavior.getActivationServiceBehavior().lookupActivations(userIds, applicationIds, timestampLastUsedBefore, timestampLastUsedAfter, activationStatus);
+            logger.info("LookupActivationsRequest succeeded");
+            return response;
+        } catch (Exception ex) {
+            logger.error("Unknown error occurred", ex);
+            throw new GenericServiceException(ServiceError.UNKNOWN_ERROR, ex.getMessage(), ex.getLocalizedMessage());
+        }
+    }
+
+    @Override
+    @Transactional
+    public UpdateStatusForActivationsResponse updateStatusForActivations(UpdateStatusForActivationsRequest request)  throws GenericServiceException {
+        if (request.getActivationIds().isEmpty()) {
+            logger.warn("Invalid request");
```

Comment:
We should:
- do a null-check on `request.getActivationIds()`
- be a bit more chatty here to hint that the issue was in missing user IDs.

Reply:
Ok, I will fix the logging in other methods as well.